src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -170,7 +162,15 @@ internal static bool IsBitwiseEquatable<T>()
 
         [Intrinsic]
         internal static ref byte GetRawSzArrayData(this Array array) =>
-            ref Unsafe.As<RawSzArrayData>(array).Data;
+            ref Unsafe.As<RawArrayData>(array).Data;
+
+        internal static unsafe ushort GetElementSize(this Array array)
+        {
```

Comment:
The inlineable managed impl will be needed to get a good performance out of managed BlockCopy impl.

Reply:
Yes, `GetObjectMethodTablePointer` is a bit of a problem.
This is just reinterpreting `object` as `IntPtr`, but whatever combination of `Unsafe` or unsafe math I use, I end up with JIT introducing a temp.
Perhaps another ` ldarg.0; conv.u; ret` intrinsic is needed.

Reply:
Yes, there is one extra instruction, but it is not a big deal.